src/me/mrletsplay/shittyauthlauncher/ShittyAuthSettingsController.java

```diff
@@ -1,10 +1,39 @@
 package me.mrletsplay.shittyauthlauncher;
 
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
 import javafx.event.ActionEvent;
 import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Node;
+import javafx.scene.Parent;
+import javafx.scene.control.Button;
 import javafx.scene.control.CheckBox;
+import javafx.scene.control.Label;
+import javafx.scene.layout.VBox;
+import me.mrletsplay.mrcore.json.JSONArray;
+import me.mrletsplay.mrcore.json.JSONObject;
+import me.mrletsplay.mrcore.misc.FriendlyException;
+import me.mrletsplay.shittyauthlauncher.util.dialog.DialogData;
+import me.mrletsplay.shittyauthlauncher.util.dialog.DialogHelper;
+import me.mrletsplay.shittyauthlauncher.util.dialog.SimpleInputDialog;
+import me.mrletsplay.shittyauthpatcher.mirrors.DownloadsMirror;
+import me.mrletsplay.shittyauthpatcher.mirrors.MojangMirror;
+import me.mrletsplay.shittyauthpatcher.version.MinecraftVersion;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
 
 public class ShittyAuthSettingsController {
+	public static ObservableList<DownloadsMirror> mirrors;
+	private ShittyAuthController controller;
```

Comment:
Maybe you should make the `settingsController` variable in the `ShittyAuth` class public to avoid having to store it here as well.